<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1241</width>
    <height>861</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QLabel" name="workArea">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>20</y>
      <width>731</width>
      <height>631</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::Shape::Box</enum>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>770</x>
      <y>20</y>
      <width>91</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>X Coordinate</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>950</x>
      <y>21</y>
      <width>91</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Y Coordinate</string>
    </property>
   </widget>
   <widget class="QLabel" name="x_coordinate">
    <property name="geometry">
     <rect>
      <x>880</x>
      <y>20</y>
      <width>63</width>
      <height>31</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::Shape::Box</enum>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="y_coordinate">
    <property name="geometry">
     <rect>
      <x>1060</x>
      <y>20</y>
      <width>63</width>
      <height>31</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::Shape::Box</enum>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QPushButton" name="showAxis">
    <property name="geometry">
     <rect>
      <x>770</x>
      <y>70</y>
      <width>83</width>
      <height>29</height>
     </rect>
    </property>
    <property name="text">
     <string>Show Axes</string>
    </property>
   </widget>
   <widget class="QPushButton" name="gridlines">
    <property name="geometry">
     <rect>
      <x>850</x>
      <y>70</y>
      <width>111</width>
      <height>29</height>
     </rect>
    </property>
    <property name="text">
     <string>Show Gridlines</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="gridOffset">
    <property name="geometry">
     <rect>
      <x>860</x>
      <y>110</y>
      <width>81</width>
      <height>29</height>
     </rect>
    </property>
    <property name="minimum">
     <number>0</number>
    </property>
    <property name="singleStep">
     <number>10</number>
    </property>
   </widget>
   <widget class="QSpinBox" name="axisWidth">
    <property name="geometry">
     <rect>
      <x>860</x>
      <y>150</y>
      <width>81</width>
      <height>29</height>
     </rect>
    </property>
    <property name="minimum">
     <number>1</number>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>770</x>
      <y>110</y>
      <width>81</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>GridOffset</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>770</x>
      <y>160</y>
      <width>81</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Axis Width</string>
    </property>
   </widget>
   <widget class="QPushButton" name="dda_line">
    <property name="geometry">
     <rect>
      <x>960</x>
      <y>70</y>
      <width>111</width>
      <height>24</height>
     </rect>
    </property>
    <property name="text">
     <string>draw dda line</string>
    </property>
   </widget>
   <widget class="QPushButton" name="draw_bressenheim_line">
    <property name="geometry">
     <rect>
      <x>940</x>
      <y>110</y>
      <width>151</width>
      <height>24</height>
     </rect>
    </property>
    <property name="text">
     <string>draw bessenheim line</string>
    </property>
   </widget>
   <widget class="QPushButton" name="draw_polar_circle">
    <property name="geometry">
     <rect>
      <x>940</x>
      <y>150</y>
      <width>121</width>
      <height>24</height>
     </rect>
    </property>
    <property name="text">
     <string>draw polar circle</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="circleRadius">
    <property name="geometry">
     <rect>
      <x>860</x>
      <y>200</y>
      <width>91</width>
      <height>25</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_5">
    <property name="geometry">
     <rect>
      <x>760</x>
      <y>200</y>
      <width>81</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Circle radius</string>
    </property>
   </widget>
   <widget class="QPushButton" name="draw_bressenham_circle">
    <property name="geometry">
     <rect>
      <x>950</x>
      <y>200</y>
      <width>161</width>
      <height>24</height>
     </rect>
    </property>
    <property name="text">
     <string>draw Bressenham circle</string>
    </property>
   </widget>
   <widget class="QLabel" name="polar_circle_time">
    <property name="geometry">
     <rect>
      <x>770</x>
      <y>240</y>
      <width>121</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Polar Circle time(ns)</string>
    </property>
   </widget>
   <widget class="QLabel" name="bressenham_circle_time">
    <property name="geometry">
     <rect>
      <x>880</x>
      <y>240</y>
      <width>151</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Bressenham Circle time(ns)</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="polartime">
    <property name="geometry">
     <rect>
      <x>770</x>
      <y>260</y>
      <width>104</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextEdit" name="bressenham_time">
    <property name="geometry">
     <rect>
      <x>900</x>
      <y>260</y>
      <width>104</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="draw_polar_ellipse">
    <property name="geometry">
     <rect>
      <x>760</x>
      <y>360</y>
      <width>151</width>
      <height>24</height>
     </rect>
    </property>
    <property name="text">
     <string>draw polar ellipse</string>
    </property>
   </widget>
   <widget class="QPushButton" name="draw_bressenham_ellipse">
    <property name="geometry">
     <rect>
      <x>910</x>
      <y>360</y>
      <width>151</width>
      <height>24</height>
     </rect>
    </property>
    <property name="text">
     <string>draw Bressenham ellipse</string>
    </property>
   </widget>
   <widget class="QLabel" name="polar_circle_time_2">
    <property name="geometry">
     <rect>
      <x>770</x>
      <y>390</y>
      <width>131</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Polar ellipse time(ns)</string>
    </property>
   </widget>
   <widget class="QLabel" name="polar_circle_time_3">
    <property name="geometry">
     <rect>
      <x>890</x>
      <y>390</y>
      <width>151</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Bressenham ellipse time(ns)</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="polar_ellipse_time">
    <property name="geometry">
     <rect>
      <x>780</x>
      <y>410</y>
      <width>104</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextEdit" name="bressenham_ellipse_time">
    <property name="geometry">
     <rect>
      <x>900</x>
      <y>410</y>
      <width>104</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QSpinBox" name="h_radius">
    <property name="geometry">
     <rect>
      <x>770</x>
      <y>320</y>
      <width>111</width>
      <height>25</height>
     </rect>
    </property>
   </widget>
   <widget class="QSpinBox" name="v_radius">
    <property name="geometry">
     <rect>
      <x>880</x>
      <y>320</y>
      <width>111</width>
      <height>25</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_6">
    <property name="geometry">
     <rect>
      <x>780</x>
      <y>300</y>
      <width>121</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Horizontal Radius</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_7">
    <property name="geometry">
     <rect>
      <x>890</x>
      <y>300</y>
      <width>121</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Vertical Radius</string>
    </property>
   </widget>
   <widget class="QPushButton" name="redraw_shape">
    <property name="geometry">
     <rect>
      <x>930</x>
      <y>460</y>
      <width>151</width>
      <height>24</height>
     </rect>
    </property>
    <property name="text">
     <string>redraw shape</string>
    </property>
   </widget>
   <widget class="QPushButton" name="flood_fill">
    <property name="geometry">
     <rect>
      <x>780</x>
      <y>500</y>
      <width>151</width>
      <height>24</height>
     </rect>
    </property>
    <property name="text">
     <string>Flood fill</string>
    </property>
   </widget>
   <widget class="QPushButton" name="draw_polygon">
    <property name="geometry">
     <rect>
      <x>780</x>
      <y>460</y>
      <width>151</width>
      <height>24</height>
     </rect>
    </property>
    <property name="text">
     <string>Draw polygon</string>
    </property>
   </widget>
   <widget class="QPushButton" name="boundary_fill">
    <property name="geometry">
     <rect>
      <x>780</x>
      <y>550</y>
      <width>151</width>
      <height>24</height>
     </rect>
    </property>
    <property name="text">
     <string>Bounday fill</string>
    </property>
   </widget>
   <widget class="QPushButton" name="fill_by_scanline">
    <property name="geometry">
     <rect>
      <x>780</x>
      <y>580</y>
      <width>151</width>
      <height>24</height>
     </rect>
    </property>
    <property name="text">
     <string>Fill by scanline</string>
    </property>
   </widget>
   <widget class="QPushButton" name="clear_canvas">
    <property name="geometry">
     <rect>
      <x>930</x>
      <y>500</y>
      <width>151</width>
      <height>24</height>
     </rect>
    </property>
    <property name="text">
     <string>Clear canvas</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="along_x">
    <property name="geometry">
     <rect>
      <x>970</x>
      <y>560</y>
      <width>81</width>
      <height>25</height>
     </rect>
    </property>
    <property name="minimum">
     <number>-338</number>
    </property>
   </widget>
   <widget class="QSpinBox" name="along_y">
    <property name="geometry">
     <rect>
      <x>1080</x>
      <y>560</y>
      <width>81</width>
      <height>25</height>
     </rect>
    </property>
    <property name="minimum">
     <number>-338</number>
    </property>
   </widget>
   <widget class="QLabel" name="label_8">
    <property name="geometry">
     <rect>
      <x>980</x>
      <y>530</y>
      <width>91</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Along x</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_9">
    <property name="geometry">
     <rect>
      <x>1090</x>
      <y>530</y>
      <width>91</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Along y</string>
    </property>
   </widget>
   <widget class="QPushButton" name="translate">
    <property name="geometry">
     <rect>
      <x>1010</x>
      <y>590</y>
      <width>80</width>
      <height>24</height>
     </rect>
    </property>
    <property name="text">
     <string>Translate</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="angle">
    <property name="geometry">
     <rect>
      <x>810</x>
      <y>620</y>
      <width>81</width>
      <height>25</height>
     </rect>
    </property>
    <property name="minimum">
     <number>0</number>
    </property>
    <property name="maximum">
     <number>360</number>
    </property>
   </widget>
   <widget class="QPushButton" name="rotate">
    <property name="geometry">
     <rect>
      <x>920</x>
      <y>620</y>
      <width>80</width>
      <height>24</height>
     </rect>
    </property>
    <property name="text">
     <string>Rotate</string>
    </property>
   </widget>
   <widget class="QCheckBox" name="checkBox">
    <property name="geometry">
     <rect>
      <x>810</x>
      <y>530</y>
      <width>91</width>
      <height>22</height>
     </rect>
    </property>
    <property name="text">
     <string>8-connected</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="shear_x">
    <property name="geometry">
     <rect>
      <x>1040</x>
      <y>640</y>
      <width>81</width>
      <height>25</height>
     </rect>
    </property>
    <property name="minimum">
     <number>0</number>
    </property>
    <property name="maximum">
     <number>360</number>
    </property>
   </widget>
   <widget class="QSpinBox" name="shear_y">
    <property name="geometry">
     <rect>
      <x>1140</x>
      <y>640</y>
      <width>81</width>
      <height>25</height>
     </rect>
    </property>
    <property name="minimum">
     <number>0</number>
    </property>
    <property name="maximum">
     <number>360</number>
    </property>
   </widget>
   <widget class="QLabel" name="label_10">
    <property name="geometry">
     <rect>
      <x>1040</x>
      <y>620</y>
      <width>51</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Shear_x</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_11">
    <property name="geometry">
     <rect>
      <x>1150</x>
      <y>620</y>
      <width>51</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Shear_y</string>
    </property>
   </widget>
   <widget class="QPushButton" name="apply_shear">
    <property name="geometry">
     <rect>
      <x>1100</x>
      <y>590</y>
      <width>80</width>
      <height>24</height>
     </rect>
    </property>
    <property name="text">
     <string>Apply shear</string>
    </property>
   </widget>
   <widget class="QPushButton" name="xaxis_reflect">
    <property name="geometry">
     <rect>
      <x>1090</x>
      <y>110</y>
      <width>81</width>
      <height>24</height>
     </rect>
    </property>
    <property name="text">
     <string>X-axis-reflect</string>
    </property>
   </widget>
   <widget class="QPushButton" name="yaxis_reflect">
    <property name="geometry">
     <rect>
      <x>1070</x>
      <y>70</y>
      <width>81</width>
      <height>24</height>
     </rect>
    </property>
    <property name="text">
     <string>Y-axis-reflect</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="scale_factor_x">
    <property name="geometry">
     <rect>
      <x>1080</x>
      <y>150</y>
      <width>81</width>
      <height>29</height>
     </rect>
    </property>
    <property name="minimum">
     <number>1</number>
    </property>
   </widget>
   <widget class="QSpinBox" name="scale_factor_y">
    <property name="geometry">
     <rect>
      <x>1120</x>
      <y>210</y>
      <width>81</width>
      <height>29</height>
     </rect>
    </property>
    <property name="minimum">
     <number>1</number>
    </property>
   </widget>
   <widget class="QLabel" name="label_12">
    <property name="geometry">
     <rect>
      <x>1080</x>
      <y>130</y>
      <width>111</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Scale factor along x</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_13">
    <property name="geometry">
     <rect>
      <x>1120</x>
      <y>190</y>
      <width>111</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Scale factor along y</string>
    </property>
   </widget>
   <widget class="QPushButton" name="scale">
    <property name="geometry">
     <rect>
      <x>1120</x>
      <y>240</y>
      <width>61</width>
      <height>24</height>
     </rect>
    </property>
    <property name="text">
     <string>Scale</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1241</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
